x/stake/keeper/validator.go

```diff
@@ -142,20 +141,16 @@ func (k Keeper) GetValidatorsByPower(ctx sdk.Context) []types.Validator {
 	validators := make([]types.Validator, maxValidators)
 	iterator := sdk.KVStoreReversePrefixIterator(store, ValidatorsByPowerIndexKey) // largest to smallest
 	i := 0
-	for {
-		if !iterator.Valid() || i > int(maxValidators-1) {
-			break
-		}
+	for ; iterator.Valid() && i < int(maxValidators); iterator.Next() {
```

Comment:
`for i := 0; iterator.Valid() && i < int(maxRetrieve[0]); iterator.Next() {`

Reply:
same, not possible because of `return validators[:i] // trim`